src/EditorFeatures/CSharpTest/OnTheFlyDocs/OnTheFlyDocsUtilitiesTests.cs

```diff
@@ -110,21 +88,620 @@ struct CustomStruct
+            
+            void UseCollection()
+            {
+                var item = collection[new CustomKey(5)];
+            }
+        }
+
+        class CustomCollection
+        {
+            public string this[CustomKey key] => key.Id.ToString();
+        }
+
+        class CustomKey
+        {
+            public int Id { get; }
+            
+            public CustomKey(int id)
+            {
+                Id = id;
+            }
+        }
+        """;
+
+        var keyText = """
+        class CustomKey
+        {
+            public int Id { get; }
+            
+            public CustomKey(int id)
+            {
+                Id = id;
+            }
+        }
+        """;
+
+        await TestSymbolContextAsync(
+            testCode,
+            (syntaxTree, semanticModel) =>
+            {
+                var classDeclaration = syntaxTree.GetRoot()
+                    .DescendantNodes()
+                    .OfType<ClassDeclarationSyntax>()
+                    .FirstOrDefault(c => c.Identifier.Text == "CustomCollection");
+
+                var indexerDeclaration = classDeclaration!.Members
+                    .OfType<IndexerDeclarationSyntax>()
+                    .First();
+
+                return semanticModel.GetDeclaredSymbol(indexerDeclaration)!;
+            },
+            keyText);
+    }
+
+    private static async Task<(Solution Solution, Document Document, SemanticModel SemanticModel, SyntaxTree SyntaxTree)>
+        SetupWorkspaceAsync(string testCode)
+    {
+        var workspace = EditorTestWorkspace.CreateCSharp(testCode);
         var solution = workspace.CurrentSolution;
         var document = solution.Projects.First().Documents.First();
 
         var syntaxTree = await document.GetSyntaxTreeAsync();
         var semanticModel = await document.GetSemanticModelAsync();
 
-        var methodInvocation = syntaxTree!.GetRoot()
+        return (solution, document, semanticModel!, syntaxTree!);
+    }
+
+    private static ISymbol GetMethodSymbol(SyntaxTree syntaxTree, SemanticModel semanticModel)
+    {
+        var methodDeclaration = syntaxTree.GetRoot()
             .DescendantNodes()
-            .OfType<InvocationExpressionSyntax>()
+            .OfType<MethodDeclarationSyntax>()
             .First();
 
-        var methodSymbol = semanticModel!.GetSymbolInfo(methodInvocation).Symbol;
+        return semanticModel.GetDeclaredSymbol(methodDeclaration)!;
+    }
+
+    private static async Task TestSymbolContextAsync(
+        string testCode,
+        Func<SyntaxTree, SemanticModel, ISymbol> getSymbol,
+        string? expectedText = null,
+        bool expectAllNull = false,
+        int resultIndex = 0)
+    {
+        var (solution, _, semanticModel, syntaxTree) = await SetupWorkspaceAsync(testCode);
+
+        var symbol = getSymbol(syntaxTree, semanticModel);
+        var result = OnTheFlyDocsUtilities.GetAdditionalOnTheFlyDocsContext(solution, symbol);
```

Comment:
probably easier to just extract this out to a helper.  Then have TestSuccessAsync and TestFailureAsync helper that uses this.  instead of trying to jam all values into 'expecteText/expectAllNull/resultIndex' parameterization approach.